lib/kafka/fetcher.rb

```diff
@@ -196,5 +207,11 @@ def fetch_batches
 
       []
     end
+
+    # An incrementing counter used to synchronize resets between the
+    # foreground and background thread.
+    def current_reset_counter
+      @current_reset_counter ||= 0
```

Comment:
The ||= operator is not thread-safe. So when calling this lazy initalizer, it is possible to face a threading issue. How about moving this to the initializer of the fetcher?

Reply:
Sounds good 👍. Updated!